### PR TITLE
Create docs project for Kedro-Viz

### PR DESCRIPTION
**Pull request by stichbury**
## Description

Part of the ongoing work to enable a documentation subproject for Kedro-Viz. See https://github.com/kedro-org/kedro-viz/issues/1524 for more details (and the [parent ticket over in the Framework project for an overview](https://github.com/kedro-org/kedro/issues/2600)).

## Development notes

Added a sphinx project with basic styling. It needs `robots.txt`, heap script, shields, styling and loads more. It has a basic information architecture, links to Kedro docs (not datasets yet) and images.

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by stichbury** [comment]
I've added myself and @astrojuanlu to this PR to work on getting it complete to build a set of Viz docs. I'm wondering if it should be against `develop` rather than `main`.

**Comment by astrojuanlu** [comment]
✨ https://kedro--1586.org.readthedocs.build/projects/kedro-viz/en/1586/

File: docs/source/conf.py
Comment (astrojuanlu) [inline docs/source/conf.py:55]: @stichbury this is what powers cross-references across different projects

File: docs/source/conf.py
Comment (astrojuanlu) [inline docs/source/conf.py:53]: This was needed for the implicit `[#heading]` targets to work

**Review by tynandebold — COMMENTED** [review]
Looks amazing so far, thank you!

A couple of comments/questions:

- You've included the custom CSS but the preview link doesn't look anything like the current Kedro docs. Do we have to do something else to make that work?
- Should we add an very short intro page, similar to [the one](https://docs.kedro.org/en/stable/introduction/index.html) Kedro has?

File: docs/source/experiment_tracking.md
Comment (stichbury) [reply docs/source/experiment_tracking.md:138]: Such a memorable rune 😆  I'm sure I'll have no trouble applying that without looking it up...after about 4 years.

**Comment by stichbury** [comment]
Yeah, I dunno. It's not a strong area of my skillset. But I did the wrong thing in making that dependency because ultimately the CSS will be down to your work on https://github.com/kedro-org/kedro/issues/3016 so we should plumb those together.

**Comment by stichbury** [comment]
> * Should we add an very short intro page, similar to [the one](https://docs.kedro.org/en/stable/introduction/index.html) Kedro has?

Yes, that would be great. We can add it into the `index.rst` rather than make a new page, perhaps. I'll make a commit and feel free to edit it.

**Comment by astrojuanlu** [comment]
> You've included the custom CSS but the preview link doesn't look anything like the current Kedro docs. Do we have to do something else to make that work?

Fixed 👍🏽

**Comment by stichbury** [comment]
OK, fixed for those comments.

I'd like to release this if possible so we can hook it up to RTD subprojects builds and have a proper `docs.kedro.org/<subfolder>` for it (if not `stable` then at least as a `latest`). 

What will this take @astrojuanlu ? Is it as simple as merging into `main`.

If we do this it means we can:

* Go back to `develop` on the Framework repo and make a branch to fix up links between Kedro docs and Viz docs
     * ...And list out redirects for when the time comes to apply them
* Test that heap is working
* Set up the CSS when https://github.com/kedro-org/kedro/issues/3016 is done
* Test https://github.com/kedro-org/kedro/pull/3168

I'd like to avoid indexing the new subproject docs for now though, so once we do start publishing, we will need to adjust the `robots.txt` for the docs subdomain @tynandebold

**Review by tynandebold — APPROVED** [review]
Fantastic!

**Review by astrojuanlu — APPROVED** [review]
From my side this is good to go! After this is merged, I'll re-enable pull request builds on RTD, and we can keep iterating

**Comment by vladimir-mck** [comment]
I've copied the navigation from kedro and added the shared css in the layout as well. The only thing that is missing is the link to dataset docs for now. Let me know if you want me to revert it.

**Comment by stichbury** [comment]
This is great @vladimir-mck and I'm merging now!

**Pull request merged**
